dev-tools/packer/xgo-scripts/before_build.sh

```diff
@@ -1,9 +1,35 @@
 #!/bin/sh
 
-PB_PATH=/go/src/github.com/elastic/packetbeat
+set -e
 
-if [ -d $PB_PATH ]; then
-	cd $PB_PATH
+BEATS_PATH=/go/src/github.com/elastic/beats
+
+# BEATNAME is in the $PACK variable
+BEATNAME=$PACK
+
+if [ $BEATNAME = "packetbeat" ]; then
+	cd $BEATS_PATH
 	patch -p1 < /gopacket_pcap.patch
-    PREFIX=/build make install-cfg
 fi
+
+
+BEAT_PATH=$BEATS_PATH/$BEATNAME
+cd $BEAT_PATH
+
+PREFIX=/build
+
+echo $PREFIX
+
+cp etc/$BEATNAME.template.json $PREFIX/$BEATNAME.template.json
+# linux
+cp $BEATNAME.yml $PREFIX/$BEATNAME-linux.yml
+# binary
+cp $BEATNAME.yml $PREFIX/$BEATNAME-binary.yml
+# darwin
+cp $BEATNAME.yml $PREFIX/$BEATNAME-darwin.yml
+# win
+cp $BEATNAME.yml $PREFIX/$BEATNAME-win.yml
+
+
+# Contains beat specific adjustments. As it is platform specific knowledge, it should be in packer not the beats itself
+PREFIX=$PREFIX make adjust-cfg
```

Comment:
The idea here, is that the knowledge inside the beat about different platforms is as minimal as possible so a new platform to compile could be added without affecting the beats. So far I didn't find a solution that on the one hand works for all beats but still allows beat specific changes. Moving all the beats specifics to packer is not an option because of the community beats. So the current state is that such adjustments are made in the adjust-cfg function. Probably a better name could be `before-build`?

@tsg @andrewkroh Any thoughts on the above?